syntax = "proto3";

package DistSSE;

service nodeRPC {

// Setup
rpc setup (SetupRequestMessage) returns ( ExecuteStatus ) {}

// Search
rpc search (SearchRequestMessage) returns (stream SearchReply) {}

// Update
rpc update (UpdateRequestMessage) returns ( ExecuteStatus ) {}

rpc batch_update (stream UpdateRequestMessage) returns ( ExecuteStatus ) {}

}

service proxyRPC{

// Setup
// rpc setup (SetupRequestMessage) returns ( ExecuteStatus ) {}

// Search
rpc search (stream SearchRequestMessage) returns (stream SearchReply) {}

// Update
rpc update (UpdateRequestMessage) returns ( ExecuteStatus ) {}

rpc batch_update (stream UpdateRequestMessage) returns ( ExecuteStatus ) {}

}

message SetupRequestMessage
{
    uint64 setup_size = 1;
    // bytes public_key = 2;
}

message SearchRequestMessage
{
    bytes enc_token = 1;
    bytes st = 2;
    // fixed32 add_count = 3;
}

message SearchReply
{
	bytes ind = 1;
	// bytes op = 2;
}

message UpdateRequestMessage
{
	uint64 node = 1;
    bytes ut = 2;
    bytes enc_value = 3;
}

message ExecuteStatus
{
	bool status = 1; 
}
